src/connectors/voice-search/connectVoiceSearch.js

```diff
@@ -0,0 +1,87 @@
+import {
+  checkRendering,
+  createDocumentationMessageGenerator,
+} from '../../lib/utils';
+import voiceSearchHelper from '../../lib/voiceSearchHelper';
+
+const withUsage = createDocumentationMessageGenerator({
+  name: 'voice-search',
+  connector: true,
+});
+
+export default function connectVoiceSearch(renderFn, unmountFn) {
+  checkRendering(renderFn, withUsage());
+
+  return (widgetParams = {}) => {
+    return {
+      init({ helper }) {
+        this._refine = (() => {
+          let previousQuery;
+          const setQueryAndSearch = query => {
+            if (query !== helper.state.query) {
+              previousQuery = helper.state.query;
+              helper.setQuery(query);
+            }
+            if (previousQuery !== undefined && previousQuery !== query) {
+              helper.search();
+            }
+          };
+
+          return setQueryAndSearch;
+        })();
+
+        this._rerender = () => {};
+
+        this._voiceSearchHelper = voiceSearchHelper({
+          onQueryChange: query => this._refine(query),
+          onStateChange: state => {
+            this._voiceListeningState = state;
```

Comment:
We could expose a `getState` on the `voiceSearchHelper` that would avoid to store the value on the instance.